weather.go

```diff
@@ -53,18 +51,17 @@ func CliOutput(token, location string) (output string) {
 	}
 	json.Unmarshal(weatherString, &r)
 
-	output = string(fmt.Sprintf("weather: %s\ncelcius: %v\n", r.OneWord, r.Celcius))
+	output = string(fmt.Sprintf("city: %s\nweather: %s\ncelcius: %v\n", r.City, r.OneWord, r.Celcius))
 	return output
 }
 
-func CallUrl(token, location string) http.Response {
+func CallUrl(token string, location *strings.Builder) http.Response {
 
 	var cu CityUnknown
 	domain := "api.openweathermap.org"
 
 	resp, err := http.Get(fmt.Sprintf("https://%s/data/2.5/weather?q=%s&appid=%s", domain, location, token))
 
-	//fail as early as possible
 	if err != nil {
 		log.Printf("an error has occured, %v", err)
```

Comment:
If there's an error, we need to bail out of the function, don't we? By all means log that it's happened, but it's not okay to just continue as though `resp` were valid, because we just found out it isn't. Sometimes just _referencing_ an invalid value could cause a panic, but even at best something weird and incorrect will happen.

Reply:
fixed